### PR TITLE
add GHA CI

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: .github/workflows/ci-julia-1.6-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    steps:
+      - name: Installing dependencies...
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y git
+      - uses: actions/checkout@v1
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
```
Comment (giordano) [inline .github/workflows/ci-julia-1.6-nightly.yml]:
```suggestion
      - name: Setup Julia
        run: |
          wget -qO - "https://julialang-s3.julialang.org/bin/linux/x64/1.6/julia-1.6.0-rc1-linux-x86_64.tar.gz" | sudo tar xzf - -C /usr/local --strip-components=1
```
This works for me

Comment (vchuravy) [reply .github/workflows/ci-julia-1.6-nightly.yml]: I mean, sure :P

File: .github/workflows/ci-julia-1.6-nightly.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    tags: '*'
+  pull_request:
+jobs:
+  CI-julia-1-6-nightly:
+    name: CI-julia-1-6-nightly
+    runs-on: ubuntu-latest
+    container:
+      image: fluxrm/flux-core:focal
+      env:
+        JULIA_DEPOT_PATH: /tmp/jl_depot
```
Comment (giordano) [inline .github/workflows/ci-julia-1.6-nightly.yml]:
```suggestion
    env:
      JULIA_DEPOT_PATH: /tmp/jl_depot
```

File: .github/workflows/ci-julia-1.6-nightly.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          sudo apt-get install -y git
+      - uses: actions/checkout@v1
+      - name: Setup Julia
+        run: |
+          wget -qO - "https://julialang-s3.julialang.org/bin/linux/x64/1.6/julia-1.6.0-rc1-linux-x86_64.tar.gz" | sudo tar xzf - -C /usr/local --strip-components=1
+      - uses: actions/cache@v1
+        env:
+          cache-name: cache-artifacts
+        with:
+          path: /tmp/jl_depot/artifacts
```
Comment (giordano) [inline .github/workflows/ci-julia-1.6-nightly.yml]:
```suggestion
          path: ${{ env.JULIA_DEPO_PATH }}/artifacts
```

**Review by giordano — COMMENTED** [review]
but this is still not writable for me....

File: .github/workflows/ci-julia-1.6-nightly.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    branches:
+      - master
+    tags: '*'
+  pull_request:
+jobs:
+  CI-julia-1-6-nightly:
+    name: CI-julia-1-6-nightly
+    runs-on: ubuntu-latest
+    container:
+      image: fluxrm/flux-core:${{ matrix.image }}
```
Comment (giordano) [inline .github/workflows/ci-julia-1.6-nightly.yml]:
```suggestion
      image: fluxrm/flux-core:${{ matrix.flux_version }}
```

File: .github/workflows/ci-julia-1.6-nightly.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - name: Julia test
+        run: |
+          julia --project=test -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd()))'
+          julia --project=test --color=yes --check-bounds=yes --code-coverage=user --depwarn=yes test/runtests.jl
```
Comment (giordano) [inline .github/workflows/ci-julia-1.6-nightly.yml]: Can also go back to
```suggestion
      - uses: julia-actions/julia-buildpkg@v1
      - uses: julia-actions/julia-runtest@v1
```

File: .github/workflows/TagBot.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+on:
+  schedule:
+    - cron: 0 * * * *
+jobs:
+  TagBot:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: JuliaRegistries/TagBot@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (giordano) [inline .github/workflows/TagBot.yml]: Old template? :stuck_out_tongue_winking_eye: 
```suggestion
name: TagBot
on:
  issue_comment:
    types:
      - created
  workflow_dispatch:
jobs:
  TagBot:
    if: github.event_name == 'workflow_dispatch' || github.actor == 'JuliaTagBot'
    runs-on: ubuntu-latest
    steps:
      - uses: JuliaRegistries/TagBot@v1
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          ssh: ${{ secrets.DOCUMENTER_KEY }}
```

Comment (vchuravy) [reply .github/workflows/TagBot.yml]: > Old template?

Yeah... I should backprop this to UCX & KA

**Comment by vchuravy** [comment]
Thanks Mose!

**Pull request merged**